src/h5web/providers/Provider.tsx

```diff
@@ -17,12 +18,26 @@ function Provider(props: Props): JSX.Element {
     return <></>;
   }
 
+  const getValue = api.getValue.bind(api);
+
   return (
     <ProviderContext.Provider
       value={{
         domain: api.domain,
         metadata,
-        getValue: api.getValue.bind(api),
+        getValue,
+        getValues: (datasetsRecord) => {
+          const dsetNames = Object.keys(datasetsRecord);
+          return Promise.all(
+            dsetNames.map((n) => getValue(datasetsRecord[n].id))
+          ).then((values) => {
+            const valuesRecord: Record<string, HDF5Value> = {};
+            dsetNames.forEach((n, i) => {
+              valuesRecord[n] = values[i];
+            });
+            return valuesRecord;
```

Comment:
You can probably simplify all this with `Object.entities` and by providing a record of dataset IDs rather than datasets -- something like this:

```ts
return Promise.all(
  Object.entities(datasetsRecord).map(([name, id]) => getValue(id).then(val => [name, val])
).then(Object.fromEntries);
```

Reply:
Oooh clever :100: